packages/ui/src/bounty/modals/ContributeFundsModal/ContributeFundsModal.tsx

```diff
@@ -48,45 +55,47 @@ export const ContributeFundsModal = () => {
     showModal,
   } = useModal<BountyContributeFundsModalCall>()
   const { api, isConnected } = useApi()
-  const minFundingLimit = api?.consts.bounty.minFundingLimit.toNumber() ?? 0
+  const [state, send] = useMachine(contributeFundsMachine)
+  const minFundingLimit = api?.consts.bounty.minFundingLimit ?? BN_ZERO
   const { active: activeMember } = useMyMemberships()
   const { allAccounts } = useMyAccounts()
-  const [amount, setAmount] = useState<string>(String(minFundingLimit))
-  const [state, send] = useMachine(contributeFundsMachine)
-  const [account, setAccount] = useState<Account>()
-  const balance = useBalance(account?.address)
+  const balance = useBalance(activeMember?.controllerAccount)
 
-  const setStakingAmount = useCallback((_, value: number) => setAmount(String(value)), [])
+  const setAmount = useCallback((amount: BN | string) => send('SET_AMOUNT', { amount: new BN(amount) }), [])
```

Comment:
There's a bunch of `BN` casting after this so shouldn't it just be:
```suggestion
  const setAmount = useCallback((amount: BN) => send('SET_AMOUNT', { amount }), [])
```